python/cudf/pyproject.toml

```diff
@@ -174,3 +130,19 @@ wheel.packages = ["cudf"]
 provider = "scikit_build_core.metadata.regex"
 input = "cudf/VERSION"
 regex = "(?P<value>.*)"
+
+[tool.ruff]
+extend = "../../pyproject.toml"
+
+[tool.ruff.lint.isort]
+combine-as-imports = true
+force-sort-within-sections = true
```

Comment:
Is there a reason that we set this to true? From a quick look at [the docs](https://docs.astral.sh/ruff/settings/#lint_isort_force-sort-within-sections) I suspect that this configuration is responsible for the vast majority of the changes in this PR. I'm not opposed to it, just wondering if we discussed it?

Reply:
This does seem to be a change from the status quo -- I overlooked this earlier, and wondered if it was just a change in ruff relative to isort. I will remove this option but we can wait for @wence- to confirm if he prefers something else before merging.

Reply:
Before: +554 −610
After: +318 −380

This reduces the diff a lot, which is good.

Reply:
I had used it in cudf-polars but I have no string opinions

Reply:
seems I misinterpreted this options with sort, thx for fixing it